src/process_utility.c

```diff
@@ -1478,13 +1478,22 @@ process_reindex(ProcessUtilityArgs *args)
 			{
 				PreventCommandDuringRecovery("REINDEX");
 				ts_hypertable_permissions_check_by_id(ht->fd.id);
-
+#if PG12_GE
+				if (stmt->concurrent)
+				{
+					stmt->concurrent = false;
+					ereport(NOTICE,
+							(errmsg("REINDEX TABLE CONCURRENTLY is unsupported on a hypertable"),
```

Comment:
It is unclear why this isn't an error. People can just drop CONCURRENTLY if they want to reindex. 

I also suggest formulating similar to this PostgreSQL error: ""concurrent index creation on system catalog tables is not supported":

```suggestion
							(errmsg("concurrent index creation on hypertables is not supported"),
```

Reply:
I am fine to throw an error here, I've seen that PostgreSQL actually also uses regular implementation in some cases and shows a NOTICE in such cases. I guess we should decide for yourselves

Reply:
Can you point to the place in the PostgreSQL code where they do this? 

It seems odd to me, because if PostgreSQL has implemented  `CONCURRENTLY` I don't understand why they wouldn't use it, unless they do to this bait-and-switch depending on object type.

Reply:
Sorry, I misunderstood when checked the code for first time, I had a different impression. Switched to produce the error as suggest. Please take a look at the updated PR info above